packages/cadl-ranch-specs/http/payload/multipart/main.tsp

```diff
@@ -330,131 +333,165 @@ namespace FormData {
+      content-type will be checked with value "application/octet-stream". Expect request:
+      ```
+      POST /upload HTTP/1.1
+      Content-Length: 428
+      Content-Type: multipart/form-data; boundary=abcde12345
+      
+      --abcde12345
+      Content-Disposition: form-data; name="id"
+      Content-Type: text/plain
+      
+      123
+      --abcde12345
+      Content-Disposition: form-data; name="address"
+      Content-Type: application/json
+      
+      {
+        "city": "X"
+      }
+      --abcde12345
+      Content-Disposition: form-data; name="profileImage"; filename="<any-name-is-ok>"
+      Content-Type: application/octet-stream
+      
+      {…file content of .jpg file…}
+      --abcde12345--
+      Content-Disposition: form-data; name="previousAddresses"
+      Content-Type: application/json
+      
+      [{
+        "city": "Y"
+      },{
+        "city": "Z"
+      }]
+      --abcde12345
+      Content-Disposition: form-data; name="pictures"; filename="<any-name-is-ok>"
+      Content-Type: application/octet-stream
+      
+      {…file content of .png file…}
+      --abcde12345
+      Content-Disposition: form-data; name="pictures"; filename="<any-name-is-ok>"
+      Content-Type: application/octet-stream
+      
+      {…file content of .png file…}
+      --abcde12345--
+      ```
+      """)
+    @doc("Test content-type: multipart/form-data for mixed scenarios")
+    @post
+    @route("/complex-parts-with-httppart")
+    op jsonArrayAndFileArrayAndOthers(
+      @header contentType: "multipart/form-data",
+      @multipartBody body: ComplexHttpPartsModelRequest,
+    ): NoContentResponse;
 
-  @scenario
-  @scenarioDoc("""
-    For File part, filename will not be checked but it is necessary otherwise cadl-ranch can't parse it;
-    content-type will be checked with value "application/octet-stream". Expect request:
-    ```
-    POST /upload HTTP/1.1
-    Content-Length: 428
-    Content-Type: multipart/form-data; boundary=abcde12345
-    
-    --abcde12345
-    Content-Disposition: form-data; name="id"
-    Content-Type: text/plain
-    
-    123
-    --abcde12345
-    Content-Disposition: form-data; name="address"
-    Content-Type: application/json
-    
-    {
-      "city": "X"
+    namespace NonString {
+      @scenario
+      @scenarioDoc("""
+        Expect request:
+        ```
+        POST /upload HTTP/1.1
+        Content-Length: 428
+        Content-Type: multipart/form-data; boundary=abcde12345
+        
+        --abcde12345
+        Content-Disposition: form-data; name="temperature"
+        Content-Type: text/plain
+        
+        0.5
+        --abcde12345
+        ```
+        """)
+      @doc("Test content-type: multipart/form-data for non string")
+      @post
+      @route("/non-string-float")
+      op float(
+        @header contentType: "multipart/form-data",
+        @multipartBody body: {
+          temperature: HttpPart<{
+            @body body: float64;
```

Comment:
just curious, do we have a RP for this?

Reply:
https://github.com/Azure/cadl-ranch/pull/484#issue-2057926153